### PR TITLE
hwmv2: port STM32L0 SoCs and boards

### PR DESCRIPTION
**Pull request by erwango**

### CODE REVIEW

File: boards/st/b_l072z_lrwan1/b_l072z_lrwan1_defconfig [hunk 1]
```diff
@@ -1,12 +1,7 @@
 # SPDX-License-Identifier: Apache-2.0
 
 # Zephyr Kernel Configuration
-CONFIG_SOC_SERIES_STM32L0X=y
-
 # Platform Configuration
```
Comment (nordicjm) [inline boards/st/b_l072z_lrwan1/b_l072z_lrwan1_defconfig:4]: Can be cleaned up in future

File: boards/dragino/dragino_nbsn95/doc/index.rst [hunk 2]
```diff
@@ -100,8 +100,7 @@ The Zephyr Dragino NBSN95 board configuration supports the following hardware fe
 Other hardware features are not yet supported on this Zephyr port.
 
 The default configuration can be found in the defconfig file:
-
-	``boards/arm/dragino_nbsn95/dragino_nbsn95_defconfig``
+:zephyr_file:`/dragino_nbsn95/dragino_nbsn95_defconfig`
```
Comment (nordicjm) [inline boards/dragino/dragino_nbsn95/doc/index.rst:103]: path is not correct

**Review by nordicjm — CHANGES_REQUESTED** [review]
Only invalid name is blocking

File: boards/st/b_l072z_lrwan1/b_l072z_lrwan1_defconfig (hunk 1 above)
Comment (erwango) [reply boards/st/b_l072z_lrwan1/b_l072z_lrwan1_defconfig]: Cleaned up as a follow up commit

**Review by nordicjm — APPROVED** [review]

**Review by gautierg-st — APPROVED** [review]

**Pull request merged**